mono/utils/mono-error-internals.h

```diff
@@ -43,10 +43,34 @@ struct _MonoErrorBoxed {
 	MonoImage *image;
 };
 
-#define error_init(error) do {	\
-	((MonoErrorInternal*)(error))->error_code = MONO_ERROR_NONE;	\
-	((MonoErrorInternal*)(error))->flags = 0;	\
-} while (0);
+// FIXME This is duplicated.
+#define ERROR_DECL \
+	MonoError error_value; \
+	MonoError * const error = &error_value; \
+	error_init_internal (error);
+
+#define ERROR_DECL_NAME(x) \
+	MonoError x; \
+	error_init_internal (&x);
```

Comment:
ok, so I think here's an issue I'm having with `ERROR_DECL_NAME`: its name parallels `ERROR_DECL` which creates a local variable `MonoError *error`.  I would expect `ERROR_DECL_NAME (my_error)` to make `MonoError *my_error`.  But it doesn't. It makes `MonoError my_error`.

Reply:
ERROR_DECL_NAME is meant to handle the vast minority case. Overwhelmingly we have "error", and we kinda want to be a pointer, and that is achieved here, with the shorter (too short?) form ERROR_DECL_NAME is custom/extended/generalized, and does less, and is used a lot less.